### DISCUSSION TITLE
Heads up on a dataclass bug that may get reported as a pydantic issue

### DISCUSSION THREAD

**Discussion by hsolbrig** [pydantic/pydantic#4032] [Question] [answered]
We encountered an issue with the  dataclass wrapper that we first thought was a pydantic issue.  It turns out that the problem is with the python dataclass library itself -- see: https://github.com/python/cpython/issues/92052  

Strictly a heads up if anyone else comes across this and reports it here.

Also - kudos on the code base!   I've learned a lot of interesting things about using python whilst pursuing this issue :-)

**Comment by samuelcolvin** [comment, marked as answer]
Thanks for the heads up.

**Comment by hsolbrig** [comment]
Unfortunately, it appears that, instead of fixing the issue, the `dataclasses` group is thinking that they want to disable the ability to copy a dataclass entirely.  If they do, it will break the code at https://github.com/samuelcolvin/pydantic/blob/master/pydantic/dataclasses.py#L159.  Hopefully we can keep that from happening, but we may have to figure out some way to treat `default_factory` entries separately.  Sigh...

**Reply by samuelcolvin** [reply]
i've replied on the issue.

**Comment by hsolbrig** [comment]
Would it be possible to have a short voice discussion on this topic?   I've submitted a [proposed fix](https://github.com/python/cpython/pull/92406), but it also requires a small change in the pydantic library itself.  I've got a draft PR but I'm running into troubles commiting it on my fork because of some of the secondary processing.  Could a) do for a bit of advice and b) want to make sure that I'm understanding your intention correctly.   Also c) would love to chat a bit about some overlaps in the pydantic and linkml projects ...
